scripts/roles.json

```diff
@@ -2895,36 +2895,6 @@
     "relatedConcepts": [
       {
         "concept": {
-          "attributes": [
-            {
-              "constraints": ["set"],
-              "name": "aria-label"
-            }
-          ],
-          "name": "form"
-        },
-        "module": "HTML"
-      },
-      {
-        "concept": {
-          "attributes": [
-            {
-              "constraints": ["set"],
-              "name": "aria-labelledby"
-            }
-          ],
-          "name": "form"
-        },
-        "module": "HTML"
-      },
-      {
-        "concept": {
-          "attributes": [
-            {
-              "constraints": ["set"],
-              "name": "name"
-            }
-          ],
           "name": "form"
         },
         "module": "HTML"
```

Comment:
I see there's `"superClass": ["landmark"]` below, but I'm not sure about that since the form shouldn't be exposed as a landmark region without an accessible name now. Do you have a suggestion about that?

Reply:
Until https://github.com/w3c/aria/issues/1764 is resolved I fear it is perhaps ambiguous as to what the expectation is here?

Even once that is resolved, if it transpires that constraints are needed on the `superclass` then the data structure as it currently stands isn't expressive enough for this afaik as it stands and we'd need a (potentially breaking) change to accomodate that conditional logic. Depending on use-cases for this library, doing such may be YAGNI?

I'm not confident/familiar enough with the concepts in specs to understand if the [superclass role](https://www.w3.org/TR/wai-aria-1.2/#superclassrole) in ARIA is relevant to the landmark exposure (or lack of) described in the AAM specs? We might be conflating an inheritance model here with the expectations on what mapping is used by Accessibility APIs (would need to understand if that Accessibility API exposure is driven by the inheritance or not).

Would need someone like @scottaohara (apologies for the out of the blue ping) to help clarify I expect.

Reply:
> [...] if it transpires that constraints are needed on the `superclass` then the data structure as it currently stands isn't expressive enough for this afaik as it stands and we'd need a (potentially breaking) change to accomodate that conditional logic.

Constraints on the `superclass` also affects the `separator` role, which either has a superclass of `widget` or `structure`, depending on whether it is focusable or not, see #587 (thanks @cmorten for linking this together)

> Depending on use-cases for this library, doing such may be YAGNI?

This causes libraries like [`eslint-plugin-jsx-a11y`](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) and [Svelte](https://svelte.dev/docs/svelte/compiler-warnings) (which rely on this library to check if a role has the `widget` superclass, to check whether something is interactive/non-interactive) to throw spurious warnings like:

```
A11y: noninteractive element cannot have nonnegative tabIndex value
A11y: Non-interactive element <div> should not be assigned mouse or keyboard event listeners.
```

But, it does seem like quite a lot of effort to fix here! It's probably easier to fix on the WAI-ARIA side of things, see https://github.com/w3c/aria/issues/1348